### PR TITLE
feat: storybook api shared

### PR DESCRIPTION
**Pull request by Kubessandra**
**Goal**:
Create a package for the client of our Storybook / stories endpoints.

**Why** ?
So we can use them in the plugin without having to duplicate the code.

Others changes.
- Upgrade node
- Upgrade turbo
- Add clean script

### CODE REVIEW

**Comment by andrico-anima** [comment]
Works great :) 

https://www.loom.com/share/01c9d96ccda34d3d9b19ee8768d92006

File: packages/storybook-api/src/storybook.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+export class StorybookApi {
+  #endpoint: string;
+  #addionalHeaders;
+  constructor(opts: StorybookApiOptions) {
+    const { storybookEndpoint, addionalHeaders } = opts;
+    this.#endpoint = storybookEndpoint;
+    this.#addionalHeaders = addionalHeaders;
+  }
+
+  getStorybookByHash = async (
```
Comment (Haimonline) [inline packages/storybook-api/src/storybook.ts:19]: For some of the endpoints, we are doing the JSON parse as part of the `StorybookApi`, and for some, we are doing it outside.
Worth aligning here; I prefer doing it as part of the API (handling an error case as well) and returning the typed result

**Review by Haimonline — COMMENTED** [review]
Looks good; left a comment regarding the json parsing

**Review by Haimonline — APPROVED** [review]

File: packages/storybook-api/src/storybook.ts (hunk 1 above)
Comment (Kubessandra) [reply packages/storybook-api/src/storybook.ts:19]: Good point, I just moved everything. 
Will make the change

**Review by andrico1234 — APPROVED** [review]

**Review by Haimonline — APPROVED** [review]

**Pull request merged**